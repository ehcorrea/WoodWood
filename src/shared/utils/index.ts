export * from './carrinho/carrinho';
export * from './responsive/responsive';
