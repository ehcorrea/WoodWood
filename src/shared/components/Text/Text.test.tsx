import { render } from '../../../test/utils';
import { Text, TextProps } from './Text';

const MOCKED_TEXT = 'Mocked Text';

const setup = (props?: TextProps) => {
  const container = render(<Text {...props}>{MOCKED_TEXT}</Text>);
  const text = container.getByText(MOCKED_TEXT);
  return { text, container };
};

describe('<Text/>', () => {
  describe('quando default redenrizado', () => {
    test('sem props customizadas', () => {
      const { container } = setup();
      expect(container).toMatchSnapshot();
    });
    test('com customSize', () => {
      const { text, container } = setup({ customSize: 30 });
      expect(text).toHaveStyle({ fontSize: 21.5 });
      expect(container).toMatchSnapshot();
    });
    test('com height', () => {
      const { text, container } = setup({ height: 30 });
      expect(text).toHaveStyle({ lineHeight: 25.5 });
      expect(container).toMatchSnapshot();
    });
  });
  test('quando Title renderizado', () => {
    const container = render(<Text.Title>{MOCKED_TEXT}</Text.Title>);
    expect(container.getByText(MOCKED_TEXT)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
  test('quando Subtitle renderizado', () => {
    const container = render(<Text.Subtitle>{MOCKED_TEXT}</Text.Subtitle>);
    expect(container.getByText(MOCKED_TEXT)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
