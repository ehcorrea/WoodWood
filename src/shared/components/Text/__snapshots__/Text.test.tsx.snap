// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text/> quando Subtitle renderizado 1`] = `
<Text
  size="large"
  style={
    [
      {
        "color": "#1A1D26",
        "fontFamily": "SofiaPro-SemiBold",
        "fontSize": 25.5,
        "lineHeight": 28.5,
      },
      undefined,
    ]
  }
  weight="semibold"
>
  Mocked Text
</Text>
`;

exports[`<Text/> quando Title renderizado 1`] = `
<Text
  size="huge"
  style={
    [
      {
        "color": "#1A1D26",
        "fontFamily": "SofiaPro-Bold",
        "fontSize": 43,
        "lineHeight": 71.5,
      },
      undefined,
    ]
  }
  weight="bold"
>
  Mocked Text
</Text>
`;

exports[`<Text/> quando default redenrizado com customSize 1`] = `
<Text
  customSize={30}
  style={
    [
      {
        "color": "#1A1D26",
        "fontFamily": "SofiaPro-Regular",
        "fontSize": 21.5,
        "lineHeight": 25.5,
      },
      undefined,
    ]
  }
>
  Mocked Text
</Text>
`;

exports[`<Text/> quando default redenrizado com height 1`] = `
<Text
  height={30}
  style={
    [
      {
        "color": "#1A1D26",
        "fontFamily": "SofiaPro-Regular",
        "fontSize": 21.5,
        "lineHeight": 25.5,
      },
      undefined,
    ]
  }
>
  Mocked Text
</Text>
`;

exports[`<Text/> quando default redenrizado sem props customizadas 1`] = `
<Text
  style={
    [
      {
        "color": "#1A1D26",
        "fontFamily": "SofiaPro-Regular",
        "fontSize": 21.5,
        "lineHeight": 25.5,
      },
      undefined,
    ]
  }
>
  Mocked Text
</Text>
`;
